src/Illuminate/Console/ConfirmableTrait.php

```diff
@@ -9,22 +9,26 @@ trait ConfirmableTrait
      *
      * This method only asks for confirmation in production.
      *
-     * @param  string  $warning
+     * @param  ?string  $warning
      * @param  \Closure|bool|null  $callback
      * @return bool
      */
-    public function confirmToProceed($warning = 'Application In Production!', $callback = null)
+    public function confirmToProceed($warning = null, $callback = null)
     {
-        $callback = is_null($callback) ? $this->getDefaultConfirmCallback() : $callback;
+        $config = $this->laravel['config'];
+        $handler = $config->get('console.confirm_handler');
```

Comment:
Service resolution should not be done via config. You should instead resolve the interface using the service container, or have a static property which is set by a service provider (like how models become aware of the event dispatcher).

Reply:
Make sense I can make that change.

Reply:
Change submittied